packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -431,13 +433,16 @@ class SliverReorderableList extends StatefulWidget {
   /// Creates a sliver list that allows the user to interactively reorder its
   /// items.
   ///
+  /// The [scrollDirection] must not be null.
+  ///
   /// The [itemCount] must be greater than or equal to zero.
   const SliverReorderableList({
     super.key,
     required this.itemBuilder,
     this.findChildIndexCallback,
     required this.itemCount,
     required this.onReorder,
+    required this.scrollDirection,
```

Comment:
I make this required because this has to be the same as the scrollview, so It feels weird to me that we let people omit this property.

Reply:
It's a bit surprising that you have to tell the sliver in which direction it scrolls (and keep that in sync manually with e.g. the ScrollView). I don't think we do this for any other sliver. Can this sliver not find out the scroll direction by e.g. looking up `Scrollable.of`?

cc @Piinks

Reply:
Actually, it looks like the scroll direction is already known, there is a _scrollDirection available in the widgets/SliverReorderableListState

Reply:
Rendering layer may be too late to apply the semantics actions to children.

Reply:
Right right, SliverReorderableList does not actually have its own rendering object anyways, my mistake. Does the _scrollDirection that is already available from the state work?

Reply:
@Piinks Do you think using Scrollable.maybeOf would be a good solution?

Reply:
oh yeah, I missed it, it looks like it was using scrollable.of. I will just use that.